src/hosting/Akka.Management.Hosting/AzureDiscoveryExtensions.cs

```diff
@@ -0,0 +1,110 @@
+        ///     and query this node. Defaults to <see cref="Dns"/>
+        /// </param>
+        /// <param name="publicPort">
+        ///     The public port of this node, usually for akka management. It will be used by other nodes to connect
+        ///     and query this node. Defaults to 8558
+        /// </param>
+        /// <returns>
+        ///     The same <see cref="ActorSystem"/> instance originally passed in.
+        /// </returns>
+        public static AkkaConfigurationBuilder WithAzureDiscovery(
+            this AkkaConfigurationBuilder builder,
+            string connectionString,
+            string serviceName = null,
+            string publicHostname = null,
+            int? publicPort = null)
+        {
+            var setup = new AzureDiscoverySetup
+            {
+                ConnectionString = connectionString,
+                ServiceName = serviceName,
+                HostName = publicHostname,
+                Port = publicPort
+            };
+            return builder.WithAzureDiscovery(setup);
+        }
+
+        /// <summary>
+        ///     Adds Akka.Discovery.Azure support to the <see cref="ActorSystem"/>.
+        ///     Note that this only adds the discovery plugin, you will still need to add ClusterBootstrap for
+        ///     a complete solution.
+        /// </summary>
+        /// <param name="builder">
+        ///     The builder instance being configured.
+        /// </param>
+        /// <param name="configure">
+        ///     An action that modifies an <see cref="AzureDiscoverySetup"/> instance, used
+        ///     to configure Akka.Discovery.Azure.
+        /// </param>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        public static AkkaConfigurationBuilder WithAzureDiscovery(
+            this AkkaConfigurationBuilder builder,
+            Action<AzureDiscoverySetup> configure)
+        {
+            var setup = new AzureDiscoverySetup();
+            configure(setup);
+            return builder.WithAzureDiscovery(setup);
+        }
+
+        /// <summary>
+        ///     Adds Akka.Discovery.Azure support to the <see cref="ActorSystem"/>.
+        ///     Note that this only adds the discovery plugin, you will still need to add ClusterBootstrap for
+        ///     a complete solution.
+        /// </summary>
+        /// <param name="builder">
+        ///     The builder instance being configured.
+        /// </param>
+        /// <param name="setup">
+        ///     The <see cref="AzureDiscoverySetup"/> instance used to configure Akka.Discovery.Azure.
+        /// </param>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        public static AkkaConfigurationBuilder WithAzureDiscovery(
```

Comment:
So one downside of having ALL of the `.Hosting` plugins in the same NuGet package is that it's going to pull in all of their dependencies too - the K8s client, the AWS client, the Azure clients.... Do you think we'd be better off taking a dependency on Akka.Hosting directly inside the Akka.Discovery.X package so it's more modular?

Reply:
I would vote for that. As a user I would not be happy with getting kubernetes and aws dependencies in my codebase when I'm deploying to azure web apps or container apps. It would not be a blocker to use though.